intermediate/01-high-level-computation-patterns.ipynb

```diff
@@ -884,25 +921,25 @@
     "For instance, grouping data by month (which have varying numbers of days) or the results of an image classification.\n",
     "The GroupBy functions are essentially a generalization of `coarsen`: \n",
     "\n",
-    "- `groupby`: divide data into distinct groups, e.g. climatologies, composites. Works when \"groups\" are exact and can be determined using equality (`==`), e.g. characters or integers. Remember that floats are not exact values.\n",
+    "- `groupby`: divide data into distinct groups, e.g. climatologies, composites. Works best when the \"group indentifiers\" or \"labels\" are exact and can be determined using equality (`==`), e.g. characters or integers. Remember that floats are not exact values.\n",
     "- `groupby_bins`: Use binning operations, e.g. histograms, to group your data.\n",
-    "- `resample`: Specialized implementation of GroupBy specifically for time grouping (so far)\n",
+    "- `resample`: Specialized implementation of GroupBy specifically for time grouping (so far), allows you to change sampling frequency of dataset.\n",
     "\n",
     "\n",
-    "```{hint}\n",
-    " Both `groupby_bins` and `resample` are implemented as `GroupBy` with a specific way of constructing group labels.\n",
+    "```{note}\n",
+    " Both `groupby_bins` and `resample` are implemented as `groupby` with a specific way of constructing group labels. The GroupBy pattern is very flexible!\n",
     "```\n",
     "\n",
     "\n",
     "### Deconstructing GroupBy\n",
     "\n",
     "The GroupBy workflow is commonly called \"split-apply-combine\".\n",
     "\n",
-    "1. \"split\" : break dataset into groups\n",
-    "1. \"apply\" : apply an operation, for instance a reduction like `mean`\n",
-    "1. \"combine\" : concatenate results from apply step along a new \"group\" dimension\n",
+    "1. *Split* : break dataset into groups\n",
+    "1. *Apply* : apply an operation, for instance a reduction like `mean`\n",
+    "1. *Combine* : concatenate results from apply step along a new \"group\" dimension\n",
     "\n",
-    "But really there is a \"hidden\" first step: identifying groups (also called factorization or binning). Usually this is the hard part.\n",
+    "But really there is a \"hidden\" first step: identifying groups (also called \"factorization\" or sometimes \"binning\"). Usually this is the hard part.\n",
     "\n",
     "In reality the workflow is: \"identify groups\" → \"split into groups\" → \"apply function\" → \"combine results\".\n"
```

Comment:
we could use [this](https://foundations.projectpythia.org/_images/xarray-split-apply-combine.jpeg) neat diagram here

![](https://foundations.projectpythia.org/_images/xarray-split-apply-combine.jpeg)

Reply:
very nice!